src/Core/Framework/Adapter/Twig/Extension/BuildBreadcrumbExtension.php

```diff
@@ -19,35 +23,50 @@ class BuildBreadcrumbExtension extends AbstractExtension
     /**
      * @internal
      *
+     * @param SalesChannelRepository<EntityCollection<SalesChannelCategoryEntity>> $salesChannelCategoryRepository
      * @param EntityRepository<CategoryCollection> $categoryRepository
      */
     public function __construct(
         private readonly CategoryBreadcrumbBuilder $categoryBreadcrumbBuilder,
-        private readonly EntityRepository $categoryRepository
+        private readonly SalesChannelRepository $salesChannelCategoryRepository,
+        private readonly EntityRepository $categoryRepository,
     ) {
     }
 
     public function getFunctions(): array
     {
+        /** @deprecated tag:v6.8.0 - Remove `needs_context` option, as the SalesChannelContext is required and the Twig Context is not needed anymore */
         return [
             new TwigFunction('sw_breadcrumb_full', $this->getFullBreadcrumb(...), ['needs_context' => true]),
             new TwigFunction('sw_breadcrumb_full_by_id', $this->getFullBreadcrumbById(...), ['needs_context' => true]),
         ];
     }
 
     /**
+     * @deprecated tag:6.8.0 - Parameter $twigContext will be removed, as it is not needed anymore
+     * @deprecated tag:6.8.0 - reason:return-type-change - Will only return `array<string, SalesChannelCategoryEntity>`
+     *
      * @param array<string, mixed> $twigContext
      *
-     * @return array<string, CategoryEntity>
+     * @return array<string, CategoryEntity|SalesChannelCategoryEntity>
      */
-    public function getFullBreadcrumb(array $twigContext, CategoryEntity $category, Context $context): array
+    public function getFullBreadcrumb(array $twigContext, CategoryEntity $category, Context|SalesChannelContext $context): array
     {
-        $salesChannel = null;
-        if (\array_key_exists('context', $twigContext) && $twigContext['context'] instanceof SalesChannelContext) {
-            $salesChannel = $twigContext['context']->getSalesChannel();
+        if (Feature::isActive('v6.8.0.0')) {
```

Comment:
the feature flag usage here is not correct. 

this would mean, the code below would only be executed, if the flag is active. but then the deprecation would never be triggered in default 6.7 environments. 

you should use the feature active check with a negation, and execute the old behaviour within this condition. then with 6.8, we can completely remove this code block and have the new behaviour without much thinking about what could removed or what need to stay. 
it is okay, if this means, we have two big branches in if/else containing much similar code.